workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/MemberReflectionUtil.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Improbable.Gdk.Core
+{
+    /// <returns>
+    ///     Helper class to list members/properties with a given attribute.
+    /// </returns>
+    public class MemberReflectionUtil
+    {
+        private readonly Type attributeType;
+
+        private const BindingFlags FieldFlags = BindingFlags.Instance | BindingFlags.NonPublic |
+                                                BindingFlags.Public | BindingFlags.DeclaredOnly;
+
+        private const BindingFlags PropertyFlags =
+            FieldFlags | BindingFlags.SetProperty | BindingFlags.GetProperty;
+
+        internal MemberReflectionUtil(Type attributeType)
+        {
+            this.attributeType = attributeType;
+        }
+
+        /// <returns>
+        ///     A MemberAdapter list of properties and fields declared in given type that match at least one of given attributes.
+        /// </returns>
+        internal List<IMemberAdapter> GetMembersWithMatchingAttributes(Type targetType)
+        {
+            List<IMemberAdapter> result = new List<IMemberAdapter>();
+            ProcessMembers(result, targetType.GetProperties(PropertyFlags), CreatePropertyAdapter);
+            ProcessMembers(result, targetType.GetFields(FieldFlags), CreateFieldAdapter);
```

Comment:
Aren't the field flags already contained in the property flags? I'm not sure how the `private const BindingFlags PropertyFlags = FieldFlags | BindingFlags.SetProperty | BindingFlags.GetProperty;` part works but it might make sense to remove the `FieldFlags` from the `PropertyFlags`.

Reply:
Indeed that seems strange, I'll try to figure out... (Code was taken as is from SDK) But it's definitely not duplicating adapters, whatever the reason is, because the tests would have caught that.

Reply:
Yeah those flags make absolutely no sense, will fix them